core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/SegmentIndexFileStore.java

```diff
@@ -289,6 +335,50 @@ public void readMergeFile(String mergeFilePath) throws IOException {
     }
   }
 
+  /**
+   * Read carbonindexmerge file and get data count
+   *
+   * @param mergeFilePath
+   * @throws IOException
+   */
+  public int readMergeIdxFileGetRowCount(String mergeFilePath) throws IOException {
```

Comment:
remove this method and please reuse `org.apache.carbondata.core.indexstore.blockletindex.SegmentIndexFileStore#readMergeFile` method and get info from there, if not direct, please i suggest to do proper refactor and get the row from there